tuist/tuist
# Add support for DocC file type

Resolves https://github.com/tuist/tuist/issues/3650

### Short description 📝

This PR adds support for the `.docc` folder file type to be added to the project. The change was made to add support for the type in https://github.com/tuist/XcodeProj/pull/660, but additional changes were required for the file type to be picked up as a valid source file.

## Changes:
- Added `docs` to the `validSourceExtensions` on a Target. 
  - **_Note:_** This could be slightly misleading considering `.docc` is a `folder` type but is a required to be a compile source for the documentation archive to be picked up. I'm open to suggestions on how to improve this part!
- Added support in the `ProjectFileElements` generation step as well (in a similar manner to how `xcassets` are added to the project)

### How to test the changes locally 🧐

- Add a documentation catalogue to your project, and reference it in the Project manifest file. Generating should include the newly created catalog in the project, and `Build Documentation` should work as expected.

**_Note:_** Would it be worth it to add a fixture for this change? 

### Checklist ✅

- [n/a] In case the PR introduces changes that affect users, the documentation has been updated.
- [n/a] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.
